templates/server-configmap.yaml

```diff
@@ -178,4 +178,16 @@ data:
 
     publicClient:
       hostPort: "{{ include "temporal.componentname" (list . "frontend") }}:{{ .Values.server.frontend.service.port }}"
+
+    dynamicConfigClient:
+      filepath: "/etc/temporal/config/dynamic_config.yaml"
+      pollInterval: "10s"
+
+  dynamic_config.yaml: |-
```

Comment:
I think the general user experience we have been targeting with helm charts is that folks don't need to clone / edit helm chart code, and that they can simply provide the configuration they need via command line parameters, e. g. 

`$ helm install --set server.config.persistence.visibility.sql.user=markmark -f mycustomsettings.yml ...`

This change is a departure from that general approach --to update dynamic configs people need to make a change to chart itself.

This is not the end of the world, just want to call this out, to make sure this is an explicit decision.

Reply:
Actually, I wonder if we can continue to allow people to specify overridable dynamic config values in values.yml and then just "source" their configuration here. Something along the lines of:


`values.yml` change:

```
...
+  dynamic_config:
+            matching.numTaskqueueReadPartitions:
+                    5
   config:
...
```

and in this file, right here:
```
dynamic_config.yaml: |-
    {{- toYaml .Values.server.dynamic_config | nindent 4 }}
```

this way, folks can simply provide dynamic config values via `--set ...` command line option or via `-f mydynamicvalues.yml`, and they don't have to edit helm chart code to change their dynamic configs.